packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,722 @@
+            port: props.port?.toString(),
+            startCommand: props.startCommand,
+            environment: props.environment,
+          },
+          configurationSource: ConfigurationSourceType.REPOSITORY,
+        },
+        repositoryUrl: props.repositoryUrl,
+        sourceCodeVersion: {
+          type: 'BRANCH',
+          value: props.branch ?? 'main',
+        },
+      },
+    };
+  }
+  /**
+   * Code from the container image.
+   */
+  public static fromImage(assets: ContainerImage, options?: ImageConfiguration): CodeConfig {
+    return {
+      ecrRepository: assets.ecrRepository,
+      imageRepository: {
+        imageIdentifier: assets.uri,
+        imageRepositoryType: assets.type,
+        imageConfiguration: options ? {
+          port: options.port,
+          environment: options.environment,
+          startCommand: options.startCommand,
+        } : {},
+      },
+    };
+  }
+}
+
+/**
+ * Describes the configuration that AWS App Runner uses to run an App Runner service
+ * using an image pulled from a source image repository.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-imageconfiguration.html
+ */
+export interface ImageConfiguration {
+  /**
+   * The port that your application listens to in the container.
+   *
+   * @default 8080
+   */
+  readonly port?: number;
+
+  /**
+   * Environment variables that are available to your running App Runner service.
+   *
+   * @default - no environment variables
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * An optional command that App Runner runs to start the application in the source image.
+   * If specified, this command overrides the Docker image’s default start command.
+   *
+   * @default - no start command
+   */
+  readonly startCommand?: string;
+}
+
+/**
+ * Describes a source image repository.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-imagerepository.html
+ */
+export interface ImageRepository {
+  /**
+   * The identifier of an image.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-imagerepository.html
+   */
+  readonly imageIdentifier: string;
+
+  /**
+   * The type of the image repository. This reflects the repository provider and whether
+   * the repository is private or public.
+   */
+  readonly imageRepositoryType: ImageRepositoryType;
+
+  /**
+   * Configuration for running the identified image.
+   */
+  readonly imageConfiguration: ImageConfiguration;
+}
+
+/**
+ * Identifies a version of code that AWS App Runner refers to within a source code repository.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-sourcecodeversion.html
+ */
+export interface SourceCodeVersion {
+  /**
+   * The type of version identifier.
+   */
+  readonly type: string;
```

Comment:
Does this have a domain of valid values?

Reply:
Yes and I have added description in the doc string to help user enter correct values.